### PR TITLE
Chore/extend linting of commits

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

PR to fix issues with proposed PR #1290, and extend to include cli and custom configs. Additional changes:

- add interactive prompt option. I originally tried with `@commitlint/prompt-cli` but didn't like it, so instead added `@commitlint/cz-commitlint` which works with the `commitizen` package

- convert js config file to ts (because why not)

-  add custom formatter to give more flexibility on output messages shown

## Git Issues

Closes #

## Screenshots/Videos
_Formatter output 1_
image

_Formatter output 2_
image

_Default output format_
image

_Interactive output_
image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on slack in the `platform-dev` channel.

### CODE REVIEW

File: commitlint.config.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      max: 'upper %d chars',
+      min: '%d chars at least',
+      emptyWarning: 'can not be empty',
+      upperLimitWarning: 'over limit',
+      lowerLimitWarning: 'below limit',
+    },
+
+    questions: {
+      type: {
+        description: "Select the type of change that you're committing:",
```
Comment (chrismclarke) [inline commitlint.config.ts:26]: I don't think these need to be made explicit (copy/paste from https://commitlint.js.org/#/reference-prompt?id=prompt),
So more for reference if we want to make changes to interactive prompts

**Comment by thisislawatts** [comment]
I think that the Circle CI step may need updating to align with the changes you've made in this branch? It's actually a helpful demonstration of another facet of this work. 

Git hooks are entirely optional for local developers and introducing a fail fast `Lint` job helps us to provide that feedback quickly to contributors who have opted out of commit hooks on their local machine. It also protects our CI from running jobs for PRs fail to meet the commit style.

File: .husky/common.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/commitizen/cz-cli/issues/627
+
+command_exists () {
+  command -v "$1" >/dev/null 2>&1
+}
+
+# Workaround for Windows 10, Git Bash and Yarn
+if command_exists winpty && test -t 1; then
+  exec < /dev/tty
+fi
```
Comment (thisislawatts) [inline .husky/common.sh:12]: Great to catch these Windows requirements 🙏🏻

**Comment by thisislawatts** [comment]
@chrismclarke I also pushed a commit into this branch to verify that the CI job fails:
https://app.circleci.com/pipelines/github/ONEARMY/community-platform/1181/workflows/55bf62f6-e5c8-4a9d-a10d-be657d3a6e9c/jobs/5393

**Review by thisislawatts — APPROVED** [review]

**Comment by chrismclarke** [comment]
Looks good to me - I'm guessing that if someone tries to make a pr that contains commits that don't meet style we can just squash-merge to clean up.

I removed `pinst` as realised that only applies if publishing package to npm. I kept finding conflicting info on whether yarn postinstall hooks actually run correctly, and the verdict I reached is yes they do (but only if package.json actually changed, and also any output gets swallowed up. testing via `"postinstall": "husky install > husky.txt",` gave correct output

**Comment by chrismclarke** [comment]
PR failing and node_module install, possibly due to legacy dependency issue (see #1409). I'm not 100% sure if it actually should be installing the full dependencies to run the check and if so whether there's a way to ignore certain errors...

**Pull request merged**